core/common/src/main/java/alluxio/security/authentication/SaslStreamClientDriver.java

```diff
@@ -101,15 +105,13 @@ public void start(String channelId) throws UnauthenticatedException, Unavailable
       throw new UnauthenticatedException(se.getMessage(), se);
     } catch (InterruptedException ie) {
       Thread.currentThread().interrupt();
-      throw new UnauthenticatedException(ie.getMessage(), ie);
+      throw new CanceledException(ie.getMessage(), ie);
     } catch (ExecutionException e) {
       Throwable cause = e.getCause();
       if (cause != null && cause instanceof StatusRuntimeException) {
```

Comment:
if cause is null this will hit an NPE on line 114

Reply:
nice catch!